NPE in BaseLifecycleManager

### Bug Report
The LCM has potential NPEs where it accessed the session instance without checking if it exists or not

```
04-14 11:39:33.955 10057 10057 E AndroidRuntime: FATAL EXCEPTION: main
04-14 11:39:33.955 10057 10057 E AndroidRuntime: java.lang.RuntimeException: Unable to stop service com.navi.android.apps.navi.sdl.SdlService@a996593: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean com.smartdevicelink.session.BaseSdlSession.removeServiceListener(com.smartdevicelink.protocol.enums.SessionType, com.smartdevicelink.protocol.ISdlServiceListener)' on a null object reference
04-14 11:39:33.955 10057 10057 E AndroidRuntime: 	at android.app.ActivityThread.handleStopService(ActivityThread.java:3592)
04-14 11:39:33.955 10057 10057 E AndroidRuntime: 	at android.app.ActivityThread.-wrap26(Unknown Source:0)
04-14 11:39:33.955 10057 10057 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1708)
04-14 11:39:33.955 10057 10057 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:105)
04-14 11:39:33.955 10057 10057 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:251)
04-14 11:39:33.955 10057 10057 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:6572)
04-14 11:39:33.955 10057 10057 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
04-14 11:39:33.955 10057 10057 E AndroidRuntime: 	at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
04-14 11:39:33.955 10057 10057 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
04-14 11:39:33.955 10057 10057 E AndroidRuntime: Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean com.smartdevicelink.session.BaseSdlSession.removeServiceListener(com.smartdevicelink.protocol.enums.SessionType, com.smartdevicelink.protocol.ISdlServiceListener)' on a null object reference
04-14 11:39:33.955 10057 10057 E AndroidRuntime: 	at com.smartdevicelink.managers.lifecycle.BaseLifecycleManager$5.removeServiceListener(Unknown Source:4)
04-14 11:39:33.955 10057 10057 E AndroidRuntime: 	at com.smartdevicelink.managers.video.VideoStreamManager.dispose(Unknown Source:20)
04-14 11:39:33.955 10057 10057 E AndroidRuntime: 	at com.smartdevicelink.managers.SdlManager.dispose(Unknown Source:43)
04-14 11:39:33.955 10057 10057 E AndroidRuntime: 	at com.navi.android.apps.navi.sdl.SdlService.onDestroy(Unknown Source:15)
04-14 11:39:33.955 10057 10057 E AndroidRuntime: 	at android.app.ActivityThread.handleStopService(ActivityThread.java:3574)
04-14 11:39:33.955 10057 10057 E AndroidRuntime: 	... 8 more
```

##### Reproduction Steps
1. Add a service listeners outside of the library
2. Disconnect from IVI
3. Notice NPE

##### Expected Behavior
No NPEs

##### Observed Behavior
NPEs

#####  Version Information
* SDL Android Version: 4.12 on because that is when the LCM was introduced into the Android library

--- a/base/src/main/java/com/smartdevicelink/managers/lifecycle/BaseLifecycleManager.java
+++ b/base/src/main/java/com/smartdevicelink/managers/lifecycle/BaseLifecycleManager.java
@@ -44,6 +44,7 @@ import com.smartdevicelink.managers.permission.PermissionManager;
 import com.smartdevicelink.marshal.JsonRPCMarshaller;
 import com.smartdevicelink.protocol.ISdlServiceListener;
 import com.smartdevicelink.protocol.ProtocolMessage;
+import com.smartdevicelink.protocol.SdlProtocolBase;
 import com.smartdevicelink.protocol.enums.FunctionID;
 import com.smartdevicelink.protocol.enums.MessageType;
 import com.smartdevicelink.protocol.enums.SessionType;
@@ -987,17 +988,30 @@ abstract class BaseLifecycleManager {
 
         @Override
         public boolean isConnected() {
-            return BaseLifecycleManager.this.session.getIsConnected();
+            synchronized (BaseLifecycleManager.this) {
+                if (BaseLifecycleManager.this.session != null) {
+                    return BaseLifecycleManager.this.session.getIsConnected();
+                }
+            }
+            return false;
         }
 
         @Override
         public void addServiceListener(SessionType serviceType, ISdlServiceListener sdlServiceListener) {
-            BaseLifecycleManager.this.session.addServiceListener(serviceType, sdlServiceListener);
+            synchronized (BaseLifecycleManager.this) {
+                if(BaseLifecycleManager.this.session != null ){
+                    BaseLifecycleManager.this.session.addServiceListener(serviceType, sdlServiceListener);
+                }
+            }
         }
 
         @Override
         public void removeServiceListener(SessionType serviceType, ISdlServiceListener sdlServiceListener) {
-            BaseLifecycleManager.this.session.removeServiceListener(serviceType, sdlServiceListener);
+            synchronized (BaseLifecycleManager.this) {
+                if (BaseLifecycleManager.this.session != null) {
+                    BaseLifecycleManager.this.session.removeServiceListener(serviceType, sdlServiceListener);
+                }
+            }
         }
 
         @Override
@@ -1064,14 +1078,25 @@ abstract class BaseLifecycleManager {
 
         @Override
         public boolean isTransportForServiceAvailable(SessionType serviceType) {
-            return BaseLifecycleManager.this.session.isTransportForServiceAvailable(serviceType);
+            synchronized (BaseLifecycleManager.this) {
+                if (BaseLifecycleManager.this.session != null) {
+                    return BaseLifecycleManager.this.session.isTransportForServiceAvailable(serviceType);
+                }
+            }
+            return false;
         }
 
         @NonNull
         @Override
         public SdlMsgVersion getSdlMsgVersion() {
-            SdlMsgVersion msgVersion = new SdlMsgVersion(rpcSpecVersion.getMajor(), rpcSpecVersion.getMinor());
-            msgVersion.setPatchVersion(rpcSpecVersion.getPatch());
+            SdlMsgVersion msgVersion;
+            if(rpcSpecVersion != null) {
+                msgVersion = new SdlMsgVersion(rpcSpecVersion.getMajor(), rpcSpecVersion.getMinor());
+                msgVersion.setPatchVersion(rpcSpecVersion.getPatch());
+            } else {
+                msgVersion = new SdlMsgVersion(1,0);
+            }
+
             return msgVersion;
         }
 
@@ -1083,7 +1108,12 @@ abstract class BaseLifecycleManager {
 
         @Override
         public long getMtu(SessionType serviceType) {
-            return BaseLifecycleManager.this.session.getMtu(serviceType);
+            synchronized (BaseLifecycleManager.this) {
+                if (BaseLifecycleManager.this.session != null) {
+                    return BaseLifecycleManager.this.session.getMtu(serviceType);
+                }
+            }
+            return SdlProtocolBase.V1_V2_MTU_SIZE;
         }
 
         @Override

